core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java

```diff
@@ -19,24 +19,47 @@
 
 package org.elasticsearch.common.geo;
 
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
 import java.util.Locale;
 
 /**
  * Enum representing the relationship between a Query / Filter Shape and indexed Shapes
  * that will be used to determine if a Document should be matched or not
  */
-public enum ShapeRelation {
+public enum ShapeRelation implements Writeable<ShapeRelation>{
 
-    INTERSECTS("intersects"),
-    DISJOINT("disjoint"),
-    WITHIN("within");
+    INTERSECTS((byte) 0, "intersects"),
```

Comment:
Sure, should I add a simple unit test to make sure the order (and ordinals) of these enum values don't change?